demo/components/modal-demo.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import {SBComponent} from 'stylable-react-component';
+import {Image, Modal} from '../../src';
+import styles from './modal-demo.st.css';
+import {RequestCloseEvent} from '../../src/components/modal/modal';
+
+export interface ModalDemoState {
+    isOpen: boolean;
+}
+
+@SBComponent(styles)
+export class ModalDemo extends React.Component<{}, ModalDemoState> {
+    public state: ModalDemoState = {
+        isOpen: false
+    };
+
+    public render() {
+        return (
+            <div>
+                <button data-automation-id="MODAL_BUTTON" onClick={this.onButtonClick}>Open The Modal!</button>
+                <Modal className="root" isOpen={this.state.isOpen} onRequestClose={this.onModalClick}>
+                    <Image role="children" className="image" />
+                </Modal>
+            </div>
+        );
+    }
+
+    private toggleOpen = () => this.setState({isOpen: !this.state.isOpen});
+
+    private onButtonClick = () => this.toggleOpen();
```

Comment:
Is there a need for this function? Why not just pass this.toggleOpen to onClick?

Reply:
Whoops, relic of the old ways. Removed